lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -66,14 +72,52 @@ def setup_encrypter(options = {})
                                 monitoring_io: mongocryptd_client_monitoring_io,
                               )
 
+        unless @encryption_options[:mongocryptd_bypass_spawn] || !mongocryptd_client_monitoring_io
```

Comment:
I think it makes sense to not spawn anything if `monitoring_io` is false on the client. Happy to discuss if you disagree.